File: Problems/Problems/Leetcode/Python/floodFill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        R, C = len(image), len(image[0])
        def fill(x, y):
            if image[x][y] == color:
                image[x][y] = newColor
                if x >= 1: fill(x-1, y)
                if x+1 < R: fill(x+1, y)
                if y >= 1: fill(x, y-1)
                if y+1 < C: fill(x, y+1)
        color = image[sr][sc]
        if color == newColor: return image
        fill(sr, sc)
        return image

File: Problems/Problems/Leetcode/C++/floodFill.cpp
class Solution {
public:
    int color;
    void fill(vector<vector<int>>& image, int x, int y, int newColor, int m, int n){
        if(x < 0 || y < 0 || x == m || y == n || image[x][y] == newColor || image[x][y] != color) return;
        image[x][y] = newColor;
        fill(image, x+1, y, newColor, m, n);
        fill(image, x, y+1, newColor, m, n);
        fill(image, x-1, y, newColor, m, n);
        fill(image, x, y-1, newColor, m, n);
    }
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int newColor) {
        int m = image.size();
        int n = image[0].size();
        color = image[sr][sc];
        if(color == newColor) return image;
        fill(image, sr, sc, newColor, m, n);
        return image;
    }
};